### PR TITLE
Fix Clang warnings - libYARP_math

### PR DESCRIPTION
**Pull request by damn1**
libYARP_math - fix file by file

- missing initialization, the content content could be garbage

### CODE REVIEW

File: src/libYARP_math/src/Quaternion.cpp [hunk 1]
```diff
@@ -17,6 +17,11 @@ class QuaternionPortContentHeader
 public:
     yarp::os::NetInt32 listTag;
     yarp::os::NetInt32 listLen;
+    QuaternionPortContentHeader()
+    {
+        listTag = 0;
+        listLen = 0;
+    }
```
Comment (drdanz) [inline src/libYARP_math/src/Quaternion.cpp]: @Nicogene What is the purpose of this? 
Can we use "QuaternionPortContentHeader() = default" instead? Or at least you should use the initialization list...

File: src/libYARP_math/src/Vec2D.cpp [hunk 2]
```diff
@@ -21,6 +21,11 @@ class Vec2DPortContentHeader
 public:
     yarp::os::NetInt32 listTag;
     yarp::os::NetInt32 listLen;
+    Vec2DPortContentHeader()
```
Comment (drdanz) [inline src/libYARP_math/src/Vec2D.cpp]: Same here...

File: src/libYARP_math/src/Quaternion.cpp (hunk 1 above)
Comment (Nicogene) [reply src/libYARP_math/src/Quaternion.cpp]: The purpose is to initialise `listTag` and `listLen`, yes we can use the initialization list.

**Comment by drdanz** [comment]
Merged, thanks.

**Pull request merged**